src/functions.php

```diff
@@ -433,3 +458,17 @@ function waterfall(array $tasks): PromiseInterface
 
     return $deferred->promise();
 }
+
+/**
+ * @internal
+ */
+function fiberMap(): FiberMap
+{
+    static $wm = null;
+
+    if ($wm === null) {
+        $wm = new FiberMap();
+    }
+
+    return $wm;
+}
```

Comment:
Initially, the idea was to use a `WeakMap`, but that turned out to be a bit more complicated than expected. Will revisit that idea later on as the current implementation might leak.

Reply:
Should we also reset the reference to this instance once the last fiber is terminated? Not sure how much memory this would leak otherwise, but might be a good idea in either case.

Reply:
Doubt memory will leak, assuming everything is handled correctly. But this is also why I want to attempt an `WeakMap` attempt. Which proofed to be tricky because it doesn't seem to accept a fiber as key.

Reply:
WeakMaps support fibers as keys, see FiberLocal in Revolt.